spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/AbstractDependsOnBeanFactoryPostProcessor.java

```diff
@@ -134,16 +134,14 @@ public int getOrder() {
 	}
 
 	private static BeanDefinition getBeanDefinition(String beanName, ConfigurableListableBeanFactory beanFactory) {
-		try {
+		if (beanFactory.containsBeanDefinition(beanName)) {
 			return beanFactory.getBeanDefinition(beanName);
 		}
-		catch (NoSuchBeanDefinitionException ex) {
-			BeanFactory parentBeanFactory = beanFactory.getParentBeanFactory();
-			if (parentBeanFactory instanceof ConfigurableListableBeanFactory) {
-				return getBeanDefinition(beanName, (ConfigurableListableBeanFactory) parentBeanFactory);
-			}
-			throw ex;
+		BeanFactory parentBeanFactory = beanFactory.getParentBeanFactory();
+		if (parentBeanFactory instanceof ConfigurableListableBeanFactory) {
+			return getBeanDefinition(beanName, ((ConfigurableListableBeanFactory) parentBeanFactory));
 		}
+		throw new NoSuchBeanDefinitionException(beanName);
```

Comment:
I am not sure what you're trying to achieve here. The context may provide an exception with additional context it has when it throws the exception (besides the `beanName`) so I am not in favour of creating such low-level exception here.

Reply:
This exception is being thrown only in case when `BeanFactory` and its parent `BeanFactory` does not contain the provided `beanName`.

Reply:
My point exactly. It's going to be the only case where it's thrown but the framework will not have created the instance, we would. And I am not keen to do that as explained in my first comment, makes sense?

Reply:
I understood but in that case, this PR becomes meaningless and it should be closed 😄